jsettlers.graphics/src/jsettlers/graphics/map/controls/original/panel/content/Label.java

```diff
@@ -20,12 +20,18 @@
 import jsettlers.graphics.utils.UIPanel;
 
 public class Label extends UIPanel {
+    public enum HorizontalAlignment {
+        LEFT,
+        CENTRE,
+        RIGHT,
+    }
 
 	private final EFontSize size;
-	private final String[] words;
+	private String[] words;
```

Comment:
You always need a point where you synchronize with the backend, since that is not aware of any UI thread scheduling. Synchronisation does not have to be here, but this is the point were it crashes.